Packs/AzureCompute/Integrations/AzureCompute_v2/AzureCompute_v2_test.py

```diff
@@ -263,6 +263,24 @@ def test_list_vms_command(mocker):
     assert command_results.to_context()["EntryContext"] == VM_LIST_EC
 
 
+def test_list_vms_command_with_pagination(requests_mock, mocker):
+    """
+    Given: Two different response mocks where one contains nextLink field and both contains two different vms.
+    When: calling list_vms_command.
+    Then: Ensures the NextLink was extracted and used correctly in the following request and that the right results were returned.
+    """
+    vms_data = load_test_data("./test_data/list_vms_command_with_pagination.json")
+    response_one = vms_data[0]
+    response_two = vms_data[1]
+    from MicrosoftApiModule import MicrosoftClient
+
+    mocker.patch.object(MicrosoftClient, "get_access_token", return_value="auth")
+    requests_mock.get("https://url/resource_group/providers/Microsoft.Compute/virtualMachines", json=response_one)
+    requests_mock.get("https://url/resource_group/providers/Microsoft.Compute/virtualMachines_page_2", json=response_two)
+    command_results = list_vms_command(client, {"resource_group": "resource_group"}, {})
+    assert command_results.to_context()["EntryContext"] == VM_LIST_EC
```

Comment:
can we add assertion for what the calls should look like on request mock? like called with x arg then y arg